src/clj/puppetlabs/services/jruby/jruby_puppet_service.clj

```diff
@@ -148,10 +149,9 @@
                         "jruby-puppet.max-active-instances.")}))
      (when (jruby-schemas/shutdown-poison-pill? pool-instance#)
        (jruby/return-instance ~jruby-service pool-instance# ~reason)
-       (sling/throw+
-        {:type    ::service-unavailable
-         :message (str "Attempted to borrow a JRuby instance from the pool "
-                       "during a shutdown. Please try again.")}))
+       (mw/throw-service-unavailable!
```

Comment:
Yeah, that seems better to me as well.  Perhaps we should make this change along with the other work in the near future to restructure those error maps.